src/coprocessor/dag/handler.rs

```diff
@@ -114,9 +114,16 @@ impl DAGRequestHandler {
             eval_cfg.set_strict_sql_mode(req.get_is_strict_sql_mode());
         }
 
-        let is_batch = enable_batch_if_possible
-            && !is_streaming
-            && super::builder::DAGBuilder::can_build_batch(req.get_executors());
+        let mut is_batch = false;
+        if enable_batch_if_possible && !is_streaming {
+            let build_batch_result =
```

Comment:
I think using `match` is more clear?

Reply:
There will be simply clippy warnings because it is a binary match (match a else b) and clippy will suggest to use the if let syntax here.